### PR TITLE
Task 11

### PR DESCRIPTION
**Pull request by josh-raftery**
Created sort by and order components which make a call to the api with queries. Added intuitive, conditional colouring of the dropdows and dynamic text. Also created a nav bar component

### CODE REVIEW

File: src/Components/Articles.jsx [hunk 1]
```diff
@@ -11,17 +11,38 @@ function Articles({topicInput}){
     const [sortByInput, setSortByInput] = useState("")
 
     useEffect(() => {
-        console.log(topicInput, ' topicInput')
-        getArticles(topicInput)
+        const queryArray = []
+        let queryString = ''
+        if(topicInput){
```
Comment (nc-mark-t) [inline src/Components/Articles.jsx:16]: Could you handle this with useSearchParams?

File: src/Components/Nav.jsx [hunk 2]
```diff
@@ -1,6 +1,42 @@
+import { Link } from "react-router-dom"
+import { UserContext } from "../contexts/User";
+import { useContext } from "react";
+
 function Nav(){
+    const { user } = useContext(UserContext)
+    const [userData] = user
     return (
         <>
```
Comment (nc-mark-t) [inline src/Components/Nav.jsx:9]: Is this fragment necessary?

File: src/api.js [hunk 3]
```diff
@@ -4,15 +4,14 @@ const api = axios.create({
   baseURL: "https://be-nc-news-ml9n.onrender.com/api",
 });
 
-function getArticles(topic) {
+function getArticles(queryString) {
   let endpoint = "/articles"
-  if(topic){
-    endpoint += `?topic=${topic}`
-  }
+  endpoint += queryString
```
Comment (nc-mark-t) [inline src/api.js:9]: Look at the axios docs for the params object, it might help you here

File: src/Components/Articles.jsx (hunk 1 above)
Comment (nc-mark-t) [reply src/Components/Articles.jsx:16]: Then you could have it reflect in the URL for the user too

**Review by nc-mark-t — COMMENTED** [review]
site is looking nice 👍

File: src/Components/Nav.jsx (hunk 2 above)
Comment (josh-raftery) [reply src/Components/Nav.jsx:9]: Good spot!

File: src/api.js (hunk 3 above)
Comment (josh-raftery) [reply src/api.js:9]: I'll look into that cheers!

**Pull request merged**